etc/spack/defaults/config.yaml

```diff
@@ -138,6 +138,20 @@ config:
   ccache: false
 
 
+  # The concretization algorithm to use in Spack. Options are:
+  #
+  #   'original': Spack's original greedy, fixed-point concretizer. This
+  #       algorithm can make decisions too early and will not backtrack
+  #       sufficiently for many specs.
+  #
+  #   'clingo': Uses a logic solver under the hood to solve DAGs with full
+  #       backtracking and optimization for user preferences.
+  #
+  # 'clingo' currently requires the clingo ASP solver to be installed and
+  # built with python bindings. 'original' is built in.
+  concretizer: original
```

Comment:
Sorry to keep bringing this up, but there's still a point that I'm confused on. My understanding of our current behavior:

- `spack install`: uses clingo if `concretizer: clingo`, otherwise original concretizer
- `spack spec`: uses original concretizer, regardless of `concretizer` setting
- `spack solve`: uses clingo, regardless of `concretizer` setting

Is this correct? If so, I think it would be better to merge `spec` and `solve` so that the behave similarly to `install`. If someone wants to explicitly try both, they can either change the setting in `config.yaml` or pass a flag on the command-line.

Reply:
> Is this correct?

`spack spec` uses the currently configured concretizer, i.e. it honors the setting in `config.yaml`.

Reply:
In that case, what's the difference between `spack spec` and `spack solve`? Can/should they be merged before this PR is merged?

Reply:
I consider `spack solve` a sort of debug command. It provides some handles to tune options that are specific to `clingo` and uses the new concretizer regardless of the setting in `config.yaml`.

Reply:
>  Can/should they be merged before this PR is merged?

I think at some point they should be merged, not sure it needs to happen in this PR.

Reply:
Okay, I'm just trying to avoid introducing a new command, then having to go through multiple release cycles of deprecating and eventually removing it.